CMakeLists.txt

```diff
@@ -6,6 +6,8 @@ if(NOT DEFINED CMAKE_MACOSX_RPATH)
   set(CMAKE_MACOSX_RPATH 0)
 endif()
 
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
@jlblancoc I think this the right way and place to do so, right?
@kartikarcot I will probably do this in a PR before this one, so this diff will disappear from this PR.

Reply:
Ok will do that

Reply:
No, you don’t have to remove it. It will be in both PRs

Reply:
Will just disappear from diff the moment i merge the other one

Reply:
`target_compile_features(${TARGET_NAME} PRIVATE cxx_std_17)` is the right way to do it I think.

Reply:
Got it

Reply:
Not sure - if there is a special CMAKE flag, will it not do the above automatically (and tailored to different compilers, if needed) ?

Reply:
> @jlblancoc I think this the right way and place to do so, right?

Yes. But I agree with @ProfFan in that I would prefer to set it as a compile feature, but I think it must be public, not private (otherwise, users not using c++17 will for sure have build errors). 
So, it would be: 

    target_compile_features(${TARGET_NAME} PUBLIC cxx_std_17)

for the target `gtsam`, and perhaps `gtsam_unstable` although the latter would be redundant since it inherits the public flags from `gtsam`.

PS: From my own testing, I found [this additional command](https://github.com/MOLAorg/mola-common/blob/5b2001fb9b0503a3e38508cf0c5839c17047dfdb/mola_cmake_functions.cmake#L117-L123) was also needed to enable C++17 on MSVC, but give it a try first with `target_compile_features()` alone since it probably was a buggy cmake/msvc version by ~2020.

Reply:
What is $TARGET_NAME supposed to be here? `gtsam` ?

Reply:
I think so

Reply:
Yes.

Reply:
Ah, of course then that line wouldn't be at the root cmake file anymore, but after defining the gtsam library target.